src/ghas_cli/utils/dependabot.py

```diff
@@ -1,36 +1,86 @@
 # -*- coding: utf-8 -*-
 #!/usr/bin/env python3
 
-from typing import Dict, List
+from typing import List
 import requests
+import json
+import time
+import logging
 
 from . import network
 
 
-def list_alerts_repo(repository: str, organization: str, token: str) -> Dict:
+def list_alerts_repo(repository: str, organization: str, token: str) -> List:
     """Get Dependabot alerts for one repository"""
 
     headers = network.get_github_headers(token)
 
     alerts_repo = []
     page = 1
     while True:
-        params = {"state": "open", "per_page": 100, "page": page}
-        alerts = requests.get(
-            url=f"https://api.github.com/repos/{organization}/{repository}/dependabot/alerts",
-            params=params,
-            headers=headers,
-        )
-        if network.check_rate_limit(alerts):
-            break
+        i = 0
+        while i < network.RETRIES:
+            params = {"state": "open", "per_page": 100, "page": page}
+            alerts = requests.get(
+                url=f"https://api.github.com/repos/{organization}/{repository}/dependabot/alerts",
+                params=params,
+                headers=headers,
+            )
+            if network.check_rate_limit(alerts):
```

Comment:
`check_rate_limits` will already sleep when necessary. Do we have a bug there?

Reply:
Indeed, good catch!